src/pages/docs/administration/migrate-spaces-with-octoterra/index.md

```diff
@@ -0,0 +1,309 @@
+
+:::div{.hint}
+The `-dummySecretVariableValues` option with can also be used when exporting projects to place a dummy value into any exported sensitive variables:
+
+```Bash
+./octoterra \
+  -url https://yourinstance.octopus.app \
+  -space Spaces-1234 \
+  -apiKey API-XXXXXXXXXXXXXXXXXXX \
+  -projectName "My Project" \
+  -lookupProjectDependencies \
+  -detachProjectTemplates \
+  -dummySecretVariableValues \
+  -dest ~/Desktop/project
+```
+
+The values of sensitive variables must be manually updated after the module is applied.
+:::
+
+## Recreating Config-as-Code projects
+
+CaC projects can not share a Git repository. When a CaC project is recreated, it must point to a new Git repository. This is done by overriding the default value of a Terraform variable in the exported module.
+
+This is an example of an exported CaC project:
+
+```hcl
+resource "octopusdeploy_project" "project_my_project" {
+  name                                 = "${var.project_my_project_name}"
+  auto_create_release                  = false
+  default_guided_failure_mode          = "EnvironmentDefault"
+  default_to_skip_if_already_installed = false
+  discrete_channel_release             = false
+  is_disabled                          = false
+  is_version_controlled                = true
+  lifecycle_id                         = "${data.octopusdeploy_lifecycles.lifecycle_application.lifecycles[0].id}"
+  project_group_id                     = "${data.octopusdeploy_project_groups.project_group_google_microservice_demo.project_groups[0].id}"
+  included_library_variable_sets       = []
+  tenanted_deployment_participation    = "${var.project_my_project_tenanted}"
+
+  git_library_persistence_settings {
+    git_credential_id  = "${data.octopusdeploy_git_credentials.gitcredential_github.git_credentials[0].id}"
+    url                = "${var.project_my_project_git_url}"
+    base_path          = "${var.project_my_project_git_base_path}"
+    default_branch     = "main"
+    protected_branches = "${jsondecode(var.project_my_project_git_protected_branches)}"
+  }
+
+  lifecycle {
+    ignore_changes = ["connectivity_policy"]
+  }
+  description = "${var.project_my_project_description_prefix}${var.project_my_project_description}${var.project_my_project_description_suffix}"
+}
+```
+
+The `git_library_persistence_settings` block defines the `url` attribute. The value of this attribute is set to the variable `project_my_project_git_url`:
+
+```hcl
+variable "project_my_project_git_url" {
+  type        = string
+  nullable    = false
+  sensitive   = false
+  description = "The git url for \"My Project\""
+  default     = "https://github.com/organization/my_project.git"
+}
+```
+
+The default value of this variable is the URL of the Git repo configured for the source project. To override this value when applying the module, pass the argument `-var=project_my_project_git_url=https://github.com/organization/new_repo_name.git`, for example:
```

Comment:
Following up on my last comment, is this still required? I have hit problems in the past when I added an existing demo repo to a new Octopus instance. i.e., it complains that the repo is already initialised. My workaround is to remove the .octopus files, commit, initialise the Octopus CaC project, and then copy the original files back. 

Could this workaround work here? I know it sucks but would it still work?

Reply:
This might be confusing so please ping me if it doesn't make sense.